rest_framework_simplejwt/views.py

```diff
@@ -28,10 +37,65 @@ def post(self, request, *args, **kwargs):
         except TokenError as e:
             raise InvalidToken(e.args[0])
 
-        return Response(serializer.validated_data, status=status.HTTP_200_OK)
+        response = Response(serializer.validated_data, status=status.HTTP_200_OK)
+
+        if api_settings.AUTH_COOKIE:
+            csrf.get_token(self.request)
+            response = self.set_auth_cookies(response, serializer.validated_data)
+
+        return response
+
+    def set_auth_cookies(self, response, data):
+        return response
 
 
-class TokenObtainPairView(TokenViewBase):
+class TokenRefreshViewBase(TokenViewBase):
+    def extract_token_from_cookie(self, request):
+        return request
+
+    def post(self, request, *args, **kwargs):
+        if api_settings.AUTH_COOKIE:
+            request = self.extract_token_from_cookie(request)
+        return super().post(request, *args, **kwargs)
+
+
+class TokenCookieViewMixin:
+    def extract_token_from_cookie(self, request):
+        token = request.COOKIES.get('{}_refresh'.format(api_settings.AUTH_COOKIE))
+        if not token:
+            raise NotAuthenticated(detail=_('Refresh cookie not set. Try to authenticate first.'))
+        else:
+            request.data['refresh'] = token
+        return request
+
+    def set_auth_cookies(self, response, data):
+        expires = self.get_refresh_token_expiration()
+        response.set_cookie(
+            api_settings.AUTH_COOKIE, data['access'],
+            expires=expires,
+            domain=api_settings.AUTH_COOKIE_DOMAIN,
+            path=api_settings.AUTH_COOKIE_PATH,
+            secure=api_settings.AUTH_COOKIE_SECURE or None,
+            httponly=True,
+            samesite=api_settings.AUTH_COOKIE_SAMESITE,
+        )
+        if 'refresh' in data:
+            response.set_cookie(
+                '{}_refresh'.format(api_settings.AUTH_COOKIE), data['refresh'],
+                expires=expires,
+                domain=None,
+                path=reverse('token_refresh'),
```

Comment:
Is it possible not to have a dependency on urls routes being named 'token_refresh' ? I am trying to use this code with djoser which depends on simplejwt and names the routes differently (https://github.com/sunscrapers/djoser/blob/master/djoser/urls/jwt.py). Maybe this is something the djoser maintainers should change. If it is not possible to change here, I can ask the djoser maintainers to change the urls route names consistent with what you have here.

Reply:
56222b8e23e1031f18953a195016c64ac2e94923